src/main/java/reposense/model/RepoLocation.java

```diff
@@ -57,29 +63,60 @@ public String getOrganization() {
     }
 
     /**
-     * Verifies {@code location} can be presented as a {@code URL} or {@code Path}.
-     * @throws InvalidLocationException if otherwise.
+     * Returns true if {@code repoArgument} is a valid local repository argument.
+     * This implementation follows directly from the {@code git clone}
+     * <a href="https://git-scm.com/docs/git-clone#_git_urls">specification</a>.
      */
-    private void verifyLocation(String location) throws InvalidLocationException {
-        boolean isValidPathLocation = false;
-        boolean isValidGitUrl = false;
-
-        try {
-            Path pathLocation = Paths.get(location);
-            isValidPathLocation = Files.exists(pathLocation);
-        } catch (InvalidPathException ipe) {
-            // Ignore exception
+    public static boolean isLocalRepo(String repoArgument) {
+        boolean containsColon = repoArgument.contains(":");
+        if (!containsColon) {
+            return true;
         }
 
-        Matcher matcher = GIT_REPOSITORY_LOCATION_PATTERN.matcher(location);
-        isValidGitUrl = matcher.matches();
+        boolean hasSlashBeforeFirstColon = repoArgument.split(":", 2)[0].contains("/");
+        if (hasSlashBeforeFirstColon) {
+            return true;
+        }
+
+        return false;
+    }
+
+    private String[] getLocalRepoNameAndOrg(String location) throws InvalidLocationException {
+        boolean isWindows = SystemUtil.isWindows();
+        if (isWindows) {
+            location = location.replaceAll("/", "\\\\");
+        }
+        Pattern localRepoPattern = isWindows
+                ? LOCAL_REPOSITORY_WINDOWS_LOCATION_PATTERN
+                : LOCAL_REPOSITORY_NON_WINDOWS_LOCATION_PATTERN;
+        Matcher localRepoMatcher = localRepoPattern.matcher(location);
 
-        if (!isValidPathLocation && !isValidGitUrl) {
-            ErrorSummary.getInstance().addErrorMessage(location, String.format(MESSAGE_INVALID_LOCATION, location));
+        if (!localRepoMatcher.matches()) {
             throw new InvalidLocationException(String.format(MESSAGE_INVALID_LOCATION, location));
         }
+
+        String tempRepoName = localRepoMatcher.group("repoName");
+        String fileSeparator = isWindows ? "\\\\" : "/";
+        String tempOrganization = localRepoMatcher.group("path").replaceAll(fileSeparator, "-");
+
+        return new String[] {tempRepoName, tempOrganization};
     }
 
+    private String[] getRemoteRepoNameAndOrg(String location) throws InvalidLocationException {
+        Matcher remoteRepoMatcher = GIT_REPOSITORY_LOCATION_PATTERN.matcher(location);
+        Matcher sshRepoMatcher = SCP_LIKE_SSH_REPOSITORY_LOCATION_PATTERN.matcher(location);
+
+        boolean isValidRemoteRepoUrl = remoteRepoMatcher.matches() || sshRepoMatcher.matches();
+        if (!isValidRemoteRepoUrl) {
+            throw new InvalidLocationException(String.format(MESSAGE_INVALID_LOCATION, location));
```

Comment:
I think we should be able to add a unit test for this.

Reply:
Thanks for the comment. However, after quite a bit of testing with the regex on my end, I realise that it is very difficult to construct an example of an invalid remote repo location that does not match either the https/ssh/git URLs and the scp-like protocol. This was added in case such a string does exist. More details on it in the comment replying to `ttps://...` invalid URL error message.

Reply:
On some further consideration, I believe the regex pattern matching can be stricter for URL matches following this [stackoverflow post](https://stackoverflow.com/questions/1547899/which-characters-make-a-url-invalid/1547940#1547940). So if it matches the format but contains illegal characters then it will throw an exception. Thanks!